source/settings.rst

```diff
@@ -808,6 +808,14 @@ The settings that are used by Git are stored in the configuration files of Git.
 
     Enter the command to be run. This can be any command that your system can run e.g. an executable program,
     a .bat script, a Python command, etc. Use the ``Browse`` button to find the command to run.
+    
+    There are some special prefixes which change the way the script is executed:
+
+    * ``plugin:`` With this prefix, the text after it is interpreted as a name of a *plugin* (refer :ref:`settings-plugins`). If a plugin with that name is found, it is run.
+    * ``navigateTo:`` (since v3.0) With this prefix, the text after it is interpreted as the actual script to run. After it finishes, the first line of its output is interpreted
+      as a commit hash and, if found, the corresponding commit is selected in the revision grid.
```

Comment:
I propose we reword it further:

```
* ``navigateTo:`` (since v3.0) The text after the prefix is interpreted as a script to run. 
After the script is executed the first line of its output is interpreted as a commit hash. 
If a commit with the matching hash is present, it is selected in the revision grid.
```

Reply:
...is interpreted as _the inline contents of_ a script to run.

You could be forgiven for thinking it's the path to a file.

Actually... is it? ;)

Reply:
Updated according to @RussKie. Sorry, @drewnoakes, I haven't understood whether you actually want the text changed.

Reply:
Drew brought an interesting point. I thought it was a file, not an inline
script. But it made me wonder....

Looking at the code I thin it is meant to be a file (or path to):

```
var revisionRef = new Executable(command, module.WorkingDir).GetOutputLines(argument).FirstOrDefault();
```

@crypto-rsa is this a correct understanding of the intent?

Reply:
Yes, it's a path. The same logic as in normal user scripts is used here. Perhaps I should change the word _script_ to _command_ so that it is consistent with the existing text above my additions?

Reply:
Ok, if it's a path, let's make that clear:

> * ``navigateTo:<script-path>`` (since v3.0) Where `<script-path>` is the path to a file containing the script to run. That script is expected to return a commit hash as the first line of its output. The UI will navigate to that commit once the script completes.

Open question: Is the path relative or absolute?

We could do the same for plugins too:

> * ``plugin:<plugin-name>`` Where `<plugin-name>` is the name of a *plugin* (refer :ref:`settings-plugins`). If a plugin with that name is found, it is run.

Reply:
It is passed into the [ProcessStartInfo.FileName](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.filename) property, which I think accepts both absolute and relative paths.

Reply:
Updated as suggested. I also removed the note about the trailing comma because the structure of the command text is now clear from the description.